examples/flows/evaluation/eval-chat-math/line_process.py

```diff
@@ -0,0 +1,67 @@
+from promptflow import tool
+
+
+@tool
+def line_process(groundtruth: str, prediction: str) -> int:
+
+    processed_result = 0
+
+    # process prediction
+    try:
+        pred_float = float(prediction) # if the prediction string is a float number, convert it into float type.
+    except Exception:
+        if '/' in prediction: # if the prediction string is a fraction, try to parse it and convert it into float type.
+            split_list = prediction.split('/') 
+            if len(split_list) == 2: # if it is a fraction with right format
+                numerator, denominator = split_list
+                try:
+                    pred_float = float(numerator) / float(denominator) 
+                except Exception:
+                    # for the wrong format with non-numeric numerator or zero denominator
+                    processed_result = -1
+            else:
+                # for the wrong format with multiple '/'
+                processed_result = -1 
+        else:
+            # for the wrong format without '/'
+            processed_result = -1
+
+    # process groundtruth

```

Comment:
Seems the process groundtruth and prediction logic are same. You can use a function to do this and just call it with different parameters

Reply:
Have used a function to solve the same logic and refine the comment. Thanks for your suggestions.

Reply:
the string_to_number function actually don't need to pass into processed_result parameter, and you don't actually need a list : If the parsing is invalid, simple return a None should be fine. If it's valid, return the float value. 

def string_to_number(raw_string: str) -> float:
    ''' Try to parse the prediction string and groundtruth string to float number. 
    Support parse int, float, fraction and recognize non-numeric string with wrong format.
    For example: '3/5', '6/10', '0.5', 'the answer is \box{2/3}', '4/7//8'
    '''
    processed_result = 0
    float_number = 0.0
    try:
        float_number = float(raw_string) 
    except Exception:
        if '/' in raw_string: 
            split_list = raw_string.split('/') 
            if len(split_list) == 2: 
                numerator, denominator = split_list
                try:
                    float_number = float(numerator) / float(denominator) 
                except Exception:
                    return None
            else:
                return None
        else:
            return None
    return float_number

pred_float  = string_to_number('2/3')
''' Early stop'''
if (pred_float is None)
    return -1
    
gt_float = string_to_number('4/7//8')
if (gt_float is None)
    return -1
    
''' both pred_float and  gt_float are valid'''
if round(pred_float, 10) == round(gt_float, 10): 
            processed_result = 1 
        else:
            processed_result = -1

Reply:
Have modified the code according to your suggestions. Thanks!